compiler-java/src/com/redhat/ceylon/ceylondoc/CeylonDocTool.java

```diff
@@ -892,8 +892,7 @@ private void copySourceFiles() throws FileNotFoundException, IOException {
                 markup.around("script src='" + getResourceUrl(pkg, "ceylondoc.js") + "' type='text/javascript'"); 
                 markup.close("head");
                 markup.open("body", "pre data-language='ceylon' style='font-family: Inconsolata, Monaco, Courier, monospace'");
-                // XXX source char encoding
-                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream()));
+                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream(), getEncoding()));
```

Comment:
Maybe use `CeylonConfig.get(DefaultToolOptions.DEFAULTS_ENCODING)` like the other code, and then fall back to utf8?

If `null` actually did occur, it would result in an NPE with existing code: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/io/InputStreamReader.java/#99